bpf/lib/conntrack.h

```diff
@@ -1043,6 +1043,23 @@ ct_has_nodeport_egress_entry4(const void *map,
 	return 0;
 }
 
+static __always_inline bool
+ct_has_nodeport_or_dsr_egress_entry4(const void *map,
+				     struct ipv4_ct_tuple *ingress_tuple)
+{
+	__u8 prev_flags = ingress_tuple->flags;
+	struct ct_entry *entry;
+
+	ingress_tuple->flags = TUPLE_F_OUT;
+	entry = map_lookup_elem(map, ingress_tuple);
+	ingress_tuple->flags = prev_flags;
+
+	if (entry)
+		return entry->node_port || entry->dsr;
+
+	return 0;
+}
+
 static __always_inline bool
 ct_has_nodeport_egress_entry6(const void *map,
```

Comment:
Why not to replace it with the new one that you have introduced where it is used and removing it completely? (same for ipv4)

Reply:
One function looks for entries where `.nodeport` is set, the other looks for entries with `.nodeport` and/or `.dsr` set. Different functionality, different usage. I don't understand how one would replace the other, can you elaborate a bit more?

Reply:
Maxim's proposal makes sense to me, especially given there are three such functions in the end (node_port, dsr, node_port or dsr).

Reply:
So my first thought was that it doesn't make much difference. But this would actually allow us to pass `or_dsr := is_defined(ENABLE_DSR)`, and that's a nice little improvement :+1:.